src/api/libcellml/generatorprofile.h

```diff
@@ -3018,6 +3017,217 @@ class LIBCELLML_EXPORT GeneratorProfile
+     */
+    std::string findRootCallString(bool forDifferentialModel) const;
+
+    /**
+     * @brief Set the @c std::string for the call to the find root method.
+     *
+     * Set the @c std::string for the call to the find root method. To be
+     * useful, the string should contain the [INDEX] tag, which will be replaced
+     * with the index of the NLA system for which we want to find the root(s).
+     *
+     * @param forDifferentialModel Whether the call to the find root method is
+     * for a differential model, as opposed to an algebraic model.
+     * @param findRootCallString The @c std::string to use for the call to the
+     * find root method.
+     */
+    void setFindRootCallString(bool forDifferentialModel,
+                               const std::string &findRootCallString);
+
+    /**
+     * @brief Get the @c std::string for the find root method.
+     *
+     * Return the @c std::string for the find root method.
+     *
+     * @param forDifferentialModel Whether the call to the find root method is
+     * for a differential model, as opposed to an algebraic model.
+     *
+     * @return The @c std::string for the find root method.
+     */
+    std::string findRootMethodString(bool forDifferentialModel) const;
+
+    /**
+     * @brief Set the @c std::string for the find root method.
+     *
+     * Set the @c std::string for the find root method. To be useful, the string
+     * should contain the [INDEX], [SIZE], and [CODE] tags, which will be
+     * replaced with the index of the NLA system for which we want to find the
+     * root(s), the size of the u array, and some code to find the root(s),
+     * respectively.
+     *
+     * @param forDifferentialModel Whether the call to the find root method is
+     * for a differential model, as opposed to an algebraic model.
+     * @param findRootMethodString The @c std::string to use for the find root
+     * method.
+     */
+    void setFindRootMethodString(bool forDifferentialModel,
+                                 const std::string &findRootMethodString);
+
+    /**
+     * @brief Get the @c std::string for the call to the NLA solve method.
+     *
+     * Return the @c std::string for the call to the NLA solve method.
+     *
+     * @param forDifferentialModel Whether the call to the NLA solve method is
+     * for a differential model, as opposed to an algebraic model.
+     *
+     * @return The @c std::string for the call to the NLA solve method.
+     */
+    std::string nlaSolveCallString(bool forDifferentialModel) const;
+
+    /**
+     * @brief Set the @c std::string for the call to the NLA solve method.
+     *
+     * Set the @c std::string for the call to the NLA solve method. To be
+     * useful, the string should contain the [INDEX] and [SIZE] tags, which will
+     * be replaced with the index of the NLA system for which we want to find
+     * the root(s) and the size of the u array, respectively.
```

Comment:
might want to define/describe the `u array`

Reply:
and the `f array` further down.

Reply:
1f02c6e0